jax/experimental/jax2tf/tests/primitives_test.py

```diff
@@ -944,5 +944,13 @@ def jax_fn_array(x):
     self.assertAllClose(tf_fn_array(np.array([3, 4, 5])),
                         np.array([4.5, 10, 17.5], jnp.bfloat16))
 
+  # test_bfloat16_array checks that https://github.com/google/jax/issues/5106 is
+  # fixed
+  def test_bfloat16_array(self):
+    @jax2tf.convert
+    def f(x):
+      return x
+    f(jnp.bfloat16(1))
```

Comment:
I am actually not able to pass this test on my machine. @jakevdp what versions of TF and NumPy are you running this with?

Reply:
Yes, it fails on my machine too, but no longer fails with an infinite recursion error.

I pushed it because ~half the jax2tf tests fail on my machine, so I was hoping it was one of the expected failures :grin:

Reply:
Oh got it! I think tests fail because you are not using a recent enough version of `tf-nightly` :-) I think the tests should run properly if you simply use the latest one. Otherwise, for future references, the [jax2tf README](https://github.com/google/jax/tree/master/jax/experimental/jax2tf) should in principle document a version that works (actually it doesn't right now, because a breaking bfloat16 change happened in October, so I will update it!).

BTW, one reason we didn't catch the error you encountered sooner is that we perform all our jax2tf tests in the TF world, i.e. with the JAX arguments converted to TF tensors with the proper dtype. The failure you encounter is a similar kind of bfloat16 bug, which I think we can work around. But to be honest, if that logic has to be removed very soon (when bfloat16 on TF and JAX are unified), it might not be worth fixing here – because verifying that all these edge cases are handled properly is probably quite hard. I would be in favor of disabling whatever problematic tests there may be in the meantime (I actually do not expect that any of our tests will fail because of your change btw, since as I mentioned, all the calls they make to jax2tf conversions use `tf.Tensor`s as parameters).

Reply:
OK, sounds good. For reference, this is the PR in question: https://github.com/google/jax/pull/5085

The jax2tf failures that are blocking the merge only show up on internal tests. I'll disable the relevant tests in order to get that change in.

Reply:
SGTM! My bad, I can not see internal failures! :)